include/seqan3/alphabet/structure/structured_rna.hpp

```diff
@@ -0,0 +1,203 @@
+//
+// ============================================================================
+
+/*!\file
+ * \ingroup structure
+ * \author Joerg Winkler <j.winkler AT fu-berlin.de>
+ * \brief Contains the composition of nucleotide with structure alphabets.
+ */
+
+#pragma once
+
+#include <iostream>
+#include <string>
+#include <utility>
+
+#include <seqan3/alphabet/composition/cartesian_composition.hpp>
+#include <seqan3/alphabet/nucleotide/concept.hpp>
+#include <seqan3/alphabet/structure/rna_structure_concept.hpp>
+
+namespace seqan3
+{
+
+/*!\brief A seqan3::cartesian_composition that joins a nucleotide alphabet with an RNA structure alphabet.
+ * \ingroup structure
+ * \tparam sequence_alphabet_t Type of the first letter; must satisfy seqan3::nucleotide_concept.
+ * \tparam structure_alphabet_t Types of further letters; must satisfy seqan3::rna_structure_concept.
+ *
+ * This composition pairs a nucleotide alphabet with a structure alphabet. The rank values
+ * correpsond to numeric values in the size of the composition, while the character values
+ * are taken from the sequence alphabet and the structure annotation is taken from the structure
+ * alphabet.
+ *
+ * As with all `seqan3::cartesian_composition` s you may access the individual alphabet letters in
+ * regular c++ tuple notation, i.e. `get<0>(t)` and objects can be brace-initialized
+ * with the individual members.
+ *
+ * ~~~~~~~~~~~~~~~{.cpp}
+ *
+ * structured_rna<rna4, dot_bracket3> l{rna4::G, dot_bracket3::PAIR_OPEN};
+ * std::cout << int(to_rank(l)) << ' '
+ *           << int(to_rank(get<0>(l))) << ' '
+ *           << int(to_rank(get<1>(l))) << '\n';
+ * // 6 2 1
+ *
+ * std::cout << to_char(l) << ' '
+ *           << to_char(get<0>(l)) << ' '
+ *           << to_char(get<1>(l)) << '\n';
+ * // G G (
+ *
+ * // modify via structured bindings and references:
+ * auto & [ seq_l, structure_l ] = l;
+ * seq_l = rna4::U;
+ * std::cout << to_char(l) << '\n';
+ * // U
+ *
+ * ~~~~~~~~~~~~~~~
+ *
+ * This seqan3::cartesian_composition itself fulfills both seqan3::nucleotide_concept and seqan3::rna_structure_concept.
+ */
+
+template <typename sequence_alphabet_t, typename structure_alphabet_t>
+      requires nucleotide_concept<sequence_alphabet_t> && rna_structure_concept<structure_alphabet_t>
+struct structured_rna :
+    public cartesian_composition<structured_rna<sequence_alphabet_t, structure_alphabet_t>,
+                                 sequence_alphabet_t, structure_alphabet_t>
+{
+    //!\brief First template parameter as member type.
+    using sequence_alphabet_type = sequence_alphabet_t;
+    //!\brief Second template parameter as member type.
+    using structure_alphabet_type = structure_alphabet_t;
+
+    //!\brief Equals the char_type of sequence_alphabet_type.
+    using char_type = underlying_char_t<sequence_alphabet_type>;
+    //!\brief Equals the char_type of the structure_alphabet_type.
+    using str_char_type = underlying_char_t<structure_alphabet_type>;
```

Comment:
same here, I would propose just removing this typedef and assign_structure and to_structure calls.